src/lexer/lexer.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <istream>
+#include <memory>
+#include <string>
+
+#include "lexer/token.h"
+
+namespace lexer {
+
+class Lexer {
+public:
+  // What type of input is given to the constructor.
+  enum class SourceTag {
+    FILE,
+    STRING,
+  };
+  Lexer(const std::string& source, SourceTag tag);
+
+  static Lexer from_file(const std::string& file) {
+    return Lexer{file, SourceTag::FILE};
```

Comment:
Curly braces initialization is commonly used when the arguments correspond exactly to the content of the object you want to construct: elements of a vector, real and imaginary part of a complex, pointer for a smart pointer.

If the arguments are only **describing** how to construct the object, then parenthesis is better.